tests/integration/ledger/ledger_test.go

```diff
@@ -0,0 +1,257 @@
+					s.FormatFlag(flags.FlagKeyAlgorithm),
+					"secp256k1",
+				})
+
+				s.Require().Error(err, "false positive, error expected")
+				s.Require().Contains(err.Error(), "")
+			})
+		})
+	})
+
+	Describe("Perform transaction signing", func() {
+		BeforeEach(func() {
+			krHome = s.T().TempDir()
+			encCfg = encoding.MakeConfig(app.ModuleBasics)
+
+			var err error
+			cmd = keys.AddKeyCommand()
+			cmd.Flags().AddFlagSet(keys.Commands("home").PersistentFlags())
+
+			mockedIn = sdktestutil.ApplyMockIODiscardOutErr(cmd)
+			mocks.RegisterGetAddressPubKeySECP256K1(s.ledger, s.accAddr, s.pubKey)
+
+			kr, clientCtx, ctx = s.NewKeyringAndCtxs(krHome, mockedIn, encCfg)
+
+			b := bytes.NewBufferString("")
+			cmd.SetOut(b)
+
+			cmd.SetArgs([]string{ledgerKey, s.FormatFlag(flags.FlagUseLedger), s.FormatFlag(flags.FlagKeyAlgorithm), "eth_secp256k1"})
+			s.Require().NoError(cmd.ExecuteContext(ctx))
+
+			keyRecord, err = kr.Key(ledgerKey)
+			s.Require().NoError(err, "can't find ledger key")
+		})
+		Context("tx bank send", func() {
+			Context("keyring execution scope", func() {
+				BeforeEach(func() {
+
+					s.ledger = mocks.NewSECP256K1(s.T())
+
+					mocks.RegisterClose(s.ledger)
+					mocks.RegisterGetPublicKeySECP256K1(s.ledger, s.pubKey)
+
+				})
+				It("should return provided to sign message", func() {
+					mocks.RegisterSignSECP256K1(s.ledger, signOkMock, nil)
+
+					ledgerAddr, err := keyRecord.GetAddress()
+					s.Require().NoError(err, "can't retirieve ledger addr from a keyring")
+
+					msg := []byte("test message")
+
+					signed, _, err := kr.SignByAddress(ledgerAddr, msg)
+					s.Require().NoError(err, "failed to sign messsage")
+					_ = signed
+
+					valid := s.pubKey.VerifySignature(msg, signed)
+					s.Require().True(valid, "invalid sigrature returned")
+
+				})
+
+				It("should raise error from ledger sign function to the top", func() {
+					mocks.RegisterSignSECP256K1Error(s.ledger)
+
+					ledgerAddr, err := keyRecord.GetAddress()
+					s.Require().NoError(err, "can't retirieve ledger addr from a keyring")
+
+					msg := []byte("test message")
+
+					_, _, err = kr.SignByAddress(ledgerAddr, msg)
+
+					s.Require().Error(err, "false positive result, error expected")
+
+					s.Require().Equal(mocks.ErrMockedSigning.Error(), err.Error(), "original and returned errors are not equal")
+				})
+
+			})
+			Context("CLI execution scope", func() {
+				BeforeEach(func() {
+					s.ledger = mocks.NewSECP256K1(s.T())
+					mocks.RegisterClose(s.ledger)
+					mocks.RegisterGetPublicKeySECP256K1(s.ledger, s.pubKey)
+
+					err := testutil.FundAccount(
+						s.ctx,
+						s.app.BankKeeper,
+						s.accAddr,
+						sdk.NewCoins(
+							sdk.NewCoin("aevmos", sdk.NewInt(100000000000000)),
+						),
+					)
+					s.Require().NoError(err)
+
+					sk, err := ethsecp256k1.GenerateKey()
+					s.Require().NoError(err)
+					receiverAccAddr, err = sdk.AccAddressFromBech32(sdk.MustBech32ifyAddressBytes("evmos", sk.PubKey().Bytes()))
```

Comment:
are you trying to generate a new address? This is using the pubkey

Reply:
yes

Reply:
you can generate it using the `tests` package